sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -159,6 +165,10 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
           Some(partitionSchema))
 
         val logicalRelation = cached.getOrElse {
+          val partitionSchemaColumnNames = partitionSchema.map(_.name.toLowerCase).toSet
+          val filteredMetastoreSchema = StructType(metastoreSchema
```

Comment:
Ah, I see. I missed that this filtering was removed in #17015. The existing ```metastoreSchema``` val can just be used then.